lib/spells/BattleSpellMechanics.cpp

```diff
@@ -629,17 +631,21 @@ void ForceFieldMechanics::setupObstacle(SpellCreatedObstacle * obstacle) const
 ///RemoveObstacleMechanics
 void RemoveObstacleMechanics::applyBattleEffects(const SpellCastEnvironment * env, const BattleSpellCastParameters & parameters, SpellCastContext & ctx) const
 {
-	if(auto obstacleToRemove = parameters.cb->battleGetObstacleOnPos(parameters.getFirstDestinationHex(), false))
+	auto obstacleToRemove = parameters.cb->battleGetAllObstaclesOnPos(parameters.getFirstDestinationHex(), false);
+	if(!obstacleToRemove.empty())
 	{
-		if(canRemove(obstacleToRemove.get(), parameters.spellLvl))
+		for(auto & i : obstacleToRemove)
 		{
 			ObstaclesRemoved obr;
-			obr.obstacles.insert(obstacleToRemove->uniqueID);
-			env->sendAndApply(&obr);
-		}
-		else
-		{
-			env->complain("Cant remove this obstacle!");
+			obr.obstacles.insert(i->uniqueID);
+			if(canRemove(i.get(), parameters.spellLvl))
+			{
+				env->sendAndApply(&obr);
+			}
+			else if(obr.obstacles.empty())
+			{
+				env->complain("Cant remove this obstacle!");
```

Comment:
It should complain only if _all_ obstacles on tile are not removable or there are no obstacles at tile.
If at least one obstacle is removable, cast is allowed (see RemoveObstacleMechanics::canBeCast) and server should not complain. 

Check for obr.obstacles.empty() should be after for loop. Also obr (and sendAndApply call) should be in outer scope.